### PR TITLE
Better Rails error handling by force_navigate

### PR DESCRIPTION
**Pull request by mfalesni**
{{pytest: -v -k widget }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 68045cb4b0635f951df50475129996b8efcbb03c:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/fixtures/pytest_selenium.py [hunk 1]
```diff
@@ -881,6 +884,15 @@ def force_navigate(page_name, _tries=0, *args, **kwargs):
...
         force_navigate("dashboard")  # Start fresh
 
+    # Same with rails errors
+    rails_e = get_rails_error()
+    if rails_e is not None:
+        logger.warning("Page was blocked by rails error, recycling.")
+        logger.error(rails_e)
+        quit()
+        kwargs.pop("start", None)
+        force_navigate("dashboard")  # Start fresh
```
Comment (seandst) [inline cfme/fixtures/pytest_selenium.py]: This doesn't normally recycle; if the rails error doesn't go away on reload (say the dashboard page is broken by a rails error) it looks like we could `force_navigate` up to the recursion limit.

Comment (seandst) [inline cfme/fixtures/pytest_selenium.py]: blocker_div seems to work the same way, which makes sense in that case, but I don't think the same technique works as well here.

Comment (mfalesni) [inline cfme/fixtures/pytest_selenium.py]: Ah right, recursion. Should protected from it.

**Comment by dajoRH** [comment]
Lint report for commit 5d9a03b1f8105107671d244c0a9c13662ac922d4:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**